tools/js-sdk-release-tools/src/mlc/apiVersion/apiVersionTypeExtractor.ts

```diff
@@ -1,135 +1,19 @@
-    });
-    return apiVersion;
-};
+import { findParametersPath, getApiVersionTypeFromOperations, getApiVersionTypeFromRestClient, tryFindRestClientPath } from "../../xlc/apiVersion/utils";
 
-// workaround for createClient function changes it's way to setup api-version
-export const findApiVersionInRestClient = (clientPath: string): string | undefined => {
-    const version2 = findApiVersionInRestClientV2(clientPath);
-    if (version2) {
-        return version2;
-    }
-    const version1 = findApiVersionInRestClientV1(clientPath);
-    return version1;
-};
-
-const getApiVersionTypeFromRestClient: IApiVersionTypeExtractor = (
+export const getApiVersionType: IApiVersionTypeExtractor = async (
     packageRoot: string
-): ApiVersionType => {
-    const clientPath = findRestClientPath(packageRoot);
-    const apiVersion = findApiVersionInRestClient(clientPath);
-    if (apiVersion && apiVersion.indexOf("-preview") >= 0)
-        return ApiVersionType.Preview;
-    if (apiVersion && apiVersion.indexOf("-preview") < 0)
-        return ApiVersionType.Stable;
-    return ApiVersionType.None;
-};
-
-const findApiVersionsInOperations = (
-    sourceFile: SourceFile | undefined
-): Array<string> | undefined => {
-    const interfaces = sourceFile?.getInterfaces();
-    const interfacesWithApiVersion = interfaces?.filter((itf) =>
-        itf.getProperty('"api-version"')
-    );
-    const apiVersions = interfacesWithApiVersion?.map((itf) => {
-        const property = itf.getMembers().filter((m) => {
-            const defaultValue = m.getChildrenOfKind(
-                SyntaxKind.StringLiteral
-            )[0];
-            return defaultValue && defaultValue.getText() === '"api-version"';
-        })[0];
-        const apiVersion = property
-            .getChildrenOfKind(SyntaxKind.LiteralType)[0]
-            .getText();
-        return apiVersion;
-    });
-    return apiVersions;
-};
-
-const getApiVersionTypeFromOperations: IApiVersionTypeExtractor = (
-    packageRoot: string
-): ApiVersionType => {
-    const paraPath = path.join(packageRoot, "src/rest/parameters.ts");
-    const sourceFile = getTsSourceFile(paraPath);
-    const apiVersions = findApiVersionsInOperations(sourceFile);
-    if (!apiVersions) return ApiVersionType.None;
-    const previewVersions = apiVersions.filter(
-        (v) => v.indexOf("-preview") >= 0
-    );
-    return previewVersions.length > 0
-        ? ApiVersionType.Preview
-        : ApiVersionType.Stable;
-};
+): Promise<ApiVersionType> => {
+    let clientPattern = "src/api/*Context.ts";
+    let typeFromClient = await getApiVersionTypeFromRestClient(packageRoot, clientPattern, tryFindRestClientPath);
+    if (typeFromClient !== ApiVersionType.None) return typeFromClient;
 
-export const getApiVersionType: IApiVersionTypeExtractor = (
-    packageRoot: string
-): ApiVersionType => {
-    const typeFromClient = getApiVersionTypeFromRestClient(packageRoot);
+    clientPattern = "src/rest/*Client.ts";
```

Comment:
is it just for back up ?

Reply:
It's use for npm package, since the package may use old emitter, and has rest client in old path. If we cannot find rest client in `src/api/*Context`, we try to search in old path.